src/cython_cmake/cmake/UseCython.cmake

```diff
@@ -140,190 +72,120 @@
-    set(_target_language "CXX")
-  endif()
-
-  if(_args_PY2)
-    set(_language_level "2")
+  set(ALL_INPUT ${CYTHON_UNPARSED_ARGUMENTS})
+  list(LENGTH ALL_INPUT INPUT_LENGTH)
+  if(NOT INPUT_LENGTH EQUAL 1)
+    message(FATAL_ERROR "One and only one input file must be specified, got '${ALL_INPUT}'")
   endif()
-  if(_args_PY3)
-    set(_language_level "3")
-  endif()
-
-  list(GET _args_UNPARSED_ARGUMENTS 0 _arg0)
-
-  # if provided, use _arg0 as the input file path
-  if(_arg0)
-    set(_source_file ${_arg0})
-
-  # otherwise, must determine source file from name, or vice versa
-  else()
-    get_filename_component(_name_ext "${_name}" EXT)
+  list(GET ALL_INPUT 0 INPUT)
 
-    # if extension provided, _name is the source file
-    if(_name_ext)
-      set(_source_file ${_name})
-      get_filename_component(_name "${_source_file}" NAME_WE)
-
-    # otherwise, assume the source file is ${_name}.pyx
+  # Set target language
+  if(NOT CYTHON_LANGUAGE)
+    get_property(_languages GLOBAL PROPERTY ENABLED_LANGUAGES)
+
+    if("C" IN_LIST _langauges AND "CXX" IN_LIST _languages)
+      # Try to compute language. Returns falsy if not found.
+      _cython_compute_language(CYTHON_LANGUAGE ${INPUT})
+      message(STATUS "${CYTHON_LANGUAGE}")
+    elseif("C" IN_LIST _languages)
+      # If only C is enabled globally, assume C
+      set(CYTHON_LANGUAGE C)
+    elseif("CXX" IN_LIST _languages)
+      # Likewise for CXX
+      set(CYTHON_LANGUAGE "CXX")
     else()
-      set(_source_file ${_name}.pyx)
+      message(FATAL_ERROR "LANGUAGE keyword required if neither C nor CXX enabled globally")
     endif()
   endif()
 
-  # Set additional flags.
-  set(_cython_args)
-  if(CYTHON_ANNOTATE)
-    list(APPEND _cython_args "--annotate")
-  endif()
-  if(CMAKE_BUILD_TYPE STREQUAL "Debug" OR
-      CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
-    list(APPEND _cython_args
-      "--gdb"
-      "--line-directives"
-    )
-  endif()
-  string(STRIP "${CYTHON_FLAGS}" _stripped_cython_flags)
-  if(_stripped_cython_flags)
-    string(REGEX REPLACE " " ";" CYTHON_FLAGS_LIST "${_stripped_cython_flags}")
-    list(APPEND _cython_args ${CYTHON_FLAGS_LIST})
+  if(NOT CYTHON_LANGUAGE)
+    message(FATAL_ERROR "LANGUAGE keyword or `# distutils: language=...` required if C and CXX are enabled globally")
+  elseif(CYTHON_LANGUAGE STREQUAL C)
+    set(language_arg "")
+    set(langauge_ext ".c")
+  elseif(CYTHON_LANGUAGE STREQUAL CXX)
+    set(language_arg "--cplus")
+    set(langauge_ext ".cxx")
+  else()
+    message(FATAL_ERROR "cython_compile_pyx LANGUAGE must be one of C or CXX")
   endif()
 
-  Cython_compile_pyx(
-    TARGET_LANGUAGE ${_target_language}
-    LANGUAGE_LEVEL ${_language_level}
-    CYTHON_ARGS ${_cython_args}
-    OUTPUT_VAR ${_args_OUTPUT_VAR}
-    ${_source_file}
-    )
+  # Place the cython files in the current binary dir if no path given
+  # Can use cmake_path for CMake 3.20+
+  if(NOT CYTHON_OUTPUT)
+    # cmake_path(GET INPUT STEM basename)
```

Comment:
I see a few commented out lines like this, I assume they can be cleaned up at this point.

Reply:
These are the CMake 3.20 versions of these commands. I guess we can just rewrite them eventually if we make it 3.20+.